src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,290 @@
+      ringParticles.push(c);
+    }
+  };
+
+  class Firework {
+    explodeSize: number;
+    firework: Particle;
+    exploded: boolean;
+    particles: Array<Particle>;
+
+    constructor() {
+      this.explodeSize = p.random(0, EXPLODE_RANGE);
+      this.firework = new Particle(p.random(p.width), p.height, true);
+      this.exploded = false;
+      this.particles = [];
+    }
+
+    update() {
+      if (!this.exploded) {
+        this.firework.applyForce(gravity);
+        this.firework.update();
+
+        if (this.firework.vel.y >= 0) {
+          this.exploded = true;
+          this.explode();
+        }
+      }
+
+      for (let i = this.particles.length - 1; i >= 0; i--) {
+        this.particles[i].applyForce(gravity);
+        this.particles[i].update();
+
+        if (this.particles[i].done()) {
+          this.particles.splice(i, 1);
+        }
+      }
+    }
+
+    done() {
+      if (this.exploded && this.particles.length === 0) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    show() {
+      if (!this.exploded) {
+        this.firework.show();
+      }
+
+      for (let i = 0; i < this.particles.length; i++) {
+        this.particles[i].show();
+      }
+    }
+
+    explode() {
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(this.firework.pos.x, this.firework.pos.y, false);
+        this.particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    firework: boolean;
+    lifespan: number;
+    acc: p5.Vector;
+    ran: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, firework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.firework = firework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this.acc = p.createVector(0, 0);
+      this.ran = p.random(0, 3);
+      if (this.firework) {
+        this.vel = p.createVector(0, p.random(-12, -8));
+      } else {
+        this.vel = p5.Vector.random2D();
+        this.vel.mult(p.random(2, 10));
+      }
+    }
+
+    applyForce(force: p5.Vector) {
+      this.acc.add(force);
+    }
+
+    update() {
+      if (!this.firework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+
+      this.vel.add(this.acc);
+      this.pos.add(this.vel);
+      this.acc.mult(0);
+    }
+
+    done() {
+      if (this.lifespan < 0) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    show() {
+      if (!this.firework) {
+        // after explosion
+        p.strokeWeight(6);
+        p.stroke(255, 211, 97, this.lifespan);
+      } else {
+        p.strokeWeight(8);
+      }
+
+      if (this.ran < 1) p.stroke('rgba(255,211,97,0.2)');
+      else if (this.ran < 2) p.stroke('rgba(255,211,97,0.4)');
+      else p.stroke('rgba(255,211,97,0.6)');
+
+      p.point(this.pos.x, this.pos.y);
+    }
+  }
+
+  class ringParticle {
+    pos: p5.Vector;
```

Comment:
please give these all `public/private`

Reply:
So I kind of went through your scenes to see what you made public/private. usually I would make public if something outside the scope of the class definition is utilizing an attribute or method. Generally it is the functions. I  do not notice a pattern or method of your choosings of private or public in your cloud, rain, cloud sections classes. Am I thinking about this right?

Reply:
I made my attempt on what I think with the private/public in the latest commit! lmk what you think.